src/components/forms/DateField/DateField.stories.mdx

```diff
@@ -0,0 +1,196 @@
+import {ArgsTable, Canvas, Meta, Story} from '@storybook/addon-docs';
+import {expect} from '@storybook/jest';
+import {userEvent, within} from '@storybook/testing-library';
+import {addDays, subDays} from 'date-fns';
+
+import {FormikDecorator} from 'story-utils/decorators';
+
+import DateField from './DateField';
+
+export const Template = args => <DateField {...args} />;
+
+<Meta
+  title="Pure React Components / Forms / DateField"
+  component={DateField}
+  decorators={[FormikDecorator]}
+  parameters={{
+    docs: {
+      source: {
+        type: 'dynamic',
+        excludeDecorators: true,
+      },
+    },
+    formik: {
+      initialValues: {
+        test: '',
+      },
+    },
+  }}
+/>
+
+## Date field
+
+We identify different types of date inputs, each coming with their most-appropriate UI element.
+
+**Nearby dates**
+
+Think of appointment dates, booking dates and similar. Here, the expected date is in the nearby
+future or past. A calendar component is appropriate because:
+
+- you can display occupied dates and/or date ranges
+- you can indicate the current date
+- it is not expected that the user needs to click the year/month button too many times
+
+**Known dates (currently not supported)**
+
+Known dates are exact dates that the user typically knows of the top of their head, such as birth
+dates, credit card expiry dates...
+
+Calendars are not suitable due to the usually bad default/initial value. Preferably they are able to
+enter the numeric value for day, month and year, leveraging autocomplete as well.
+
+On mobile devices, you'd typically want to use the native date widget too.
+
+### Datepicker/calendar
+
+The datepicker/calendar is suitable for nearby dates. We aim to make it accessible to assistive
+technology and keyboard-only navigation.
+
+Known issues/enhancements:
+
+- Use arrow keys to navigate inside calendar
+- Selecting a date in the next/previous month has some odd behaviour at the moment
+
```

Comment:
Impressive! 
Works with Tab as well as with Spacebar in Chromatic Storybook for me in all browsers (viewing https://634043ae879dc17405b0e567-rntplqvorq.chromatic.com/?path=/story/pure-react-components-forms-datefield--datepicker). 
I don't see odd behavior in the selection of next/previous month? 
Additionally: I don't see a visual difference for disabled dates?
 
 
<img width="1286" alt="Screenshot 2023-05-31 at 18 48 24" src="https://github.com/open-formulieren/open-forms-sdk/assets/118177951/170d4d74-d1be-4fd0-aa37-3e6a3b4b04a5">

Reply:
as remarked in the office - styling is borked now because of the pending design tokens PR :laughing:

Reply:
> I don't see odd behavior in the selection of next/previous month?

I do.